src/main/java/com/mongodb/spark/sql/connector/write/MongoDataWriterFactory.java

```diff
@@ -31,7 +36,26 @@
  * will be created on executors and do the actual writing. So this interface must be serializable
  * and {@link MongoDataWriter} doesn't need to be.
  */
-public class MongoDataWriterFactory implements DataWriterFactory, Serializable {
+class MongoDataWriterFactory
+    implements DataWriterFactory, StreamingDataWriterFactory, Serializable {
```

Comment:
[nit]
Let's not duplicate the `Serializable` interface, `DataWriterFactory` is already `Serializable`.

Reply:
Done